packages/frontend-shared/src/gql-components/topnav/LoginModal.vue

```diff
@@ -103,18 +129,52 @@ fragment LoginModal on Query {
 }
 `
 
+gql`
+mutation LoginModal_ResetAuthState {
+  resetAuthState {
+    ...Auth
+  }
+}
+`
+
+const resetAuth = useMutation(LoginModal_ResetAuthStateDocument)
+
 const setIsOpen = (value: boolean) => {
   emit('update:modelValue', value)
+  if (!value) {
+    resetAuth.executeMutation({})
+  }
 }
 const { t } = useI18n()
 
 const viewer = computed(() => props.gql?.cloudViewer)
+const errorType = computed(() => {
+  const { name } = props.gql.authState
+
+  if (name === 'AUTH_COULD_NOT_LAUNCH_BROWSER') {
+    return 'browserError'
+  }
+
+  if (name === 'AUTH_ERROR_DURING_LOGIN') {
+    return 'loginError'
+  }
+
+  return null
```

Comment:
Can this ever be null? Also, is there any other potential errors? Should we be more strict, something like:

```ts
if (!name) {
  return null
}

if (name !== 'AUTH_COULD_NOT_LAUNCH_BROWSER' && name !== 'AUTH_ERROR_DURING_LOGIN') {
  throw Error(`Unexpected error message: ${name}`)
}

return name
```

This would force us to handle any other authentication error (eg, maybe we add a new one in the future).

Reply:
I see why this is confusing. `packages/server/lib/gui/auth.ts` since, before this PR, uses `packages/server/lib/errors.js` to grab the message content for the `AUTH_BROWSER_LAUNCHED` and `AUTH_COULD_NOT_LAUNCH_BROWSER` situations. Only one of them is an actual error. I added `AUTH_ERROR_DURING_LOGIN` which is a new catch, previously such errors would have been unhandled, or captured in the general error handling flow. But other name values would not strictly be errors.

I actually think we can completely stop relying on the `errors.getMsgByType(name, arg1)` stuff since for these states, we actually don't care about a formatted error message. You can only get these errors in the GUI, and since our text is coming from i18n, we're only really needing the `message` field to provide things like the login URL, or the actual error message, nothing that needs to be interpolated and formatted on the server any more.

Gonna clean that up a bit, it'll make this PR simpler.